### PR TITLE
WIP - Fix rounding error when pricelist show discount + fixed price

### PR DESCRIPTION
**Pull request by rdeodoo**
As long as the ratio between the product base price and the product fixed discount price has over 2 digits, those digits are lost, thus making the price subtotal compute wrong since it is QTY * Price * discount

Possible fix for #46417

### CODE REVIEW

**Comment by rdeodoo** [comment]
@JKE-be What is your opinion on this problem?
This is about fixing issue 46417. This code is just a test, it surely is perf killer but it isolate the issue (see this [video](https://drive.google.com/file/d/1UtHGAG8XtUyuO36agUMFrXhBZgW2NEOY/view)).

Ideal fix in master would probably be to just do change the `discount` field precision to store the decimals:
```python
discount = fields.Float(string='Discount (%)', digits='Discount', default=0.0)
```
to
```python
discount = fields.Float(string='Discount (%)', default=0.0)
```
and then round it in templates (eg SO pdf) or have a rounded field to use in templates
```python
discount_rounded = fields.Float(compute='_compute_discount_rounded', string='Discount Rounded (%)', default=0.0)

def _compute_discount_rounded(self):
    for order in self:
        order.discount_rounded = order.discount
```

**Comment by rdeodoo** [comment]
@Feyensv Hello,

Could I have your input here? Did you already see/discuss this bug?

After discuss with @JKE-be his opinion is the following, and I kinda agree with it:
When selecting show discount on a pricelist, it is most of the times to use a discount, not a fixed price, as it is weird to show to the user a discount of 16.63%.
Fixing it shouldn't impact perfs or it wouldn't be worth it.

That's probably why this issue exists in every version and hasn't been fixed so far, we could consider this as a corner case.

File: addons/sale/models/sale.py [hunk 1]
```diff
@@ -1038,7 +1038,7 @@ def _compute_amount(self):
         Compute the amounts of the SO line.
         """
         for line in self:
-            price = line.price_unit * (1 - (line.discount or 0.0) / 100.0)
+            price = line.price_unit * (1 - (line.discount_not_rounded or 0.0) / 100.0)
```
Comment (Feyensv) [inline addons/sale/models/sale.py:1041]: You do not support the manually specified discounts with this field ;).

**Comment by rdeodoo** [comment]
See https://github.com/odoo/odoo/issues/46417#issuecomment-719555396

**Pull request closed without merging**